da6462-assign1/crypto.py

```diff
@@ -0,0 +1,258 @@
+"""
+Assignment 1: Cryptography
+Course: CS 92SI
+Name: <YOUR NAME>
+Date: <DATE>
+
+Replace this with a description of the program.
+"""
+
+def encrypt_caesar(plaintext):
+    """
+    Encrypts plaintext using a Caesar cipher.
+    Add more implementation details here.
+    """
+    encrypt = []
+    for i in plaintext.upper():
+        c = ord(i) + 3
+        if c > 90:
+            encrypt.append(chr(c - 26))
+        else:
+            encrypt.append(chr(c))
+    return "".join(encrypt)
+
+def decrypt_caesar(ciphertext):
+    """
+    Decrypts a ciphertext using a Caesar cipher.
+    Add more implementation details here.
+    """
+    decrypt = []
+    for i in ciphertext.upper():
+        c = ord(i) - 3
+        if c < 65:
+            decrypt.append(chr(c + 26))
+        else:
+            decrypt.append(chr(c))
+    return "".join(decrypt)
+
+def encrypt_vigenere(plaintext, keyword):
+    """
+    Encrypts plaintext using a Vigenere cipher with a keyword.
+    Add more implementation details here.
+    """
+    keyword = (keyword *((len(plaintext) / len(keyword)) + 1)).upper()
+    keyword = keyword[: len(plaintext)]
+    final = []
+    for i, c in enumerate(plaintext.upper()):
+        letter = ord(c) + ord(keyword[i])
+        while letter > 90 or letter < 65:
+            temp = letter % 91
+            temp = letter % 26
+            letter = 65 + temp
+        final.append(chr(letter))
+    return "".join(final)
+
+def decrypt_vigenere(ciphertext, keyword):
+    """
+    Decrypts ciphertext using a Vigenere cipher with a keyword.
+    Add more implementation details here.
+    """
+    keyword = (keyword *((len(ciphertext) / len(keyword)) + 1)).upper()
+    keyword = keyword[: len(ciphertext)]
+    final = []
+    for i, c in enumerate(ciphertext.upper()):
+        letter = ord(c) - ord(keyword[i])
+        while letter > 90 or letter < 65:
+            temp = letter % 91
+            temp = letter % 26
+            letter = 65 + temp
+        final.append(chr(letter))
+    return "".join(final)
+
+def encrypt_railfence(plaintext, num_rails):
+    """
+    Encrypts plaintext using a railfence cipher.
+    Add more implementation details here.
+    """
+    total_chr = len(plaintext)
+    direction = 1
+    zig_zag = [""] *num_rails
+    rail = 0
+    i = 0
+    while total_chr > 0:
+        zig_zag[rail] += plaintext[i]
+        total_chr -= 1
+        if rail == num_rails - 1 and direction == 1:
+            direction *= -1
+        if rail == 0 and direction == -1:
+            direction *= -1
```

Comment:
It looks like you can combine these two cases above - since you're inverting the direction either way, it might be cleaner to do:
```
if rail == num_rails - 1 or rail == 0:
    direction *= -1
```